django/core/management/commands/makemessages.py

```diff
@@ -349,11 +349,14 @@ def handle(self, *args, **options):
             self.default_locale_path = self.locale_paths[0]
             self.invoked_for_django = True
         else:
+            # Settings are available if run from project root
             if self.settings_available:
                 self.locale_paths.extend(settings.LOCALE_PATHS)
-            # Allow to run makemessages inside an app dir
-            if os.path.isdir('locale'):
+            # if settings aren't available, or locale path exists we use the locale path
+            if os.path.isdir('locale') or not self.settings_available:
                 self.locale_paths.append(os.path.abspath('locale'))
+
```

Comment:
nitpick: this blank line steals some thunder from the more important blank line below IMO